File: abc/python/abc219/abc219_B.py

import sys
infile = sys.stdin.buffer

def gs()  : return infile.readline().rstrip()
def gi()  : return int(gs())
def gf()  : return float(gs())
def gss() : return gs().split()
def gis() : return [int(x) for x in gss()]
def gfs() : return [float(x) for x in gss()]

def main(infn="") :
    global infile
    infile = open(infn,"r") if infn else open(sys.argv[1],"r") if len(sys.argv) > 1 else sys.stdin
    S1 = gs(); S2 = gs(); S3 = gs(); T = gs()
    ansarr = []
    for c in T : ansarr.append(S1 if c == '1' else S2 if c == '2' else S3)
    ans = "".join(ansarr)
    sys.stdout.write(str(ans)+'\n')

if __name__ == '__main__' :
    main()
    sys.stdout.flush()

File: abc/go/abc219/abc219_B/abc219_B.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"os"
	"strings"
)

type scanner struct { sc *bufio.Scanner }
func newScanner(input io.Reader) *scanner {
	sc := bufio.NewScanner(input)
	sc.Split(bufio.ScanWords)
	sc.Buffer(make([]byte, 1024), int(1e+9))
	return &scanner{sc}	
}
var rdr = newScanner(os.Stdin)
const BUFSIZE = 10000000
var wrtr = bufio.NewWriterSize(os.Stdout, BUFSIZE)
func gs() string  { rdr.sc.Scan(); return rdr.sc.Text() }
func main() {
	//f1, _ := os.Create("cpu.prof"); pprof.StartCPUProfile(f1); defer pprof.StopCPUProfile()
    defer wrtr.Flush()
	infn := ""
	if infn == "" && len(os.Args) > 1 {	infn = os.Args[1] }
    if infn != "" {	f, e := os.Open(infn); if e != nil { panic(e) }; rdr = newScanner(f) }
	S1,S2,S3,T := gs(),gs(),gs(),gs()
	ansarr := make([]string,len(T))
	for i,c := range T { 
		if c == '1' { ansarr[i] = S1 }
		if c == '2' { ansarr[i] = S2 }
		if c == '3' { ansarr[i] = S3 }
	}
	ans := strings.Join(ansarr,"")
	fmt.Println(ans)
}



